commonTools/build_stats/BuildStatsWrappers.cmake

```diff
@@ -8,6 +8,92 @@ set(BUILD_STATS_SRC_DIR "${CMAKE_CURRENT_LIST_DIR}")
+  
+  #  should ideally ask for the dtypes or suitable regexes to vet them
+  foreach(flag ${GNU_TIME_POSSIBLE_FIELDS})
+    message(DEBUG "----------------------")
+    message(DEBUG "Time: Testing field ${flag}")
+    # the output from time goes to stderr, the programs output to stdout
+    set(GNU_TIME_RC "-1")
+    set(GNU_TIME_OUTPUT "")
+    execute_process(COMMAND "${GNU_TIME_EXE}"
+                              "--format=%${flag}" "true"
+                    # this is useless - we run a noop command
+                    RESULT_VARIABLE GNU_TIME_RC
+                    # capture stderr
+                    ERROR_VARIABLE GNU_TIME_OUTPUT
+                    )
+    # if this fails, then something is broken on the system.
+    # the checks after will likely fail, because they expect a predefined
+    # format for stderr text
+    if(GNU_TIME_RC MATCHES "0")
+      #message("Time invocation error returned `${GNU_TIME_RC}` but expected `0`")
+    else()
+      message(DEBUG "Time invocation error returned `${GNU_TIME_RC}` but expected `0`")
+      message("-- GNU_TIME_EXE=${GNU_TIME_EXE} does not work")
+      message("-- Set BUILD_STATS_TIME_CMD if /usr/bin/time is invalid")
+      unset(BUILD_STATS_TIME_CMD CACHE)
+      return()
+    endif()
+
+    # this isn't ideal, we match the failure mode, but we should match valid output
+    # That is harder to do, as we would need to determine expected format for each
+    # command (e.g., dtype or a suitable regex from the python script)
+    # A good todo would be to have the pythong script report headeres it will
+    # try and a valid expected format regex, then we test against that here.
+    if("${GNU_TIME_OUTPUT}" MATCHES "^?${flag}.*")
+      message(DEBUG "Time does not support Field: ${flag}")
+      # not ideal, but let's quit if we don't find the fields...
+      # a better solution would be to disable the field, and only
+      # error out if we can't find some required set of fields.
+      message("-- GNU_TIME_EXE=${GNU_TIME_EXE} does not work")
+      message("-- Set BUILD_STATS_TIME_CMD if /usr/bin/time is invalid")
+      unset(BUILD_STATS_TIME_CMD CACHE)
+      return()
+    else()
+      message(DEBUG "Time supports Field: ${flag}")
+      list(APPEND GNU_TIME_SUPPORTED_FIELDS "${flag}")
+    endif()
+  endforeach()
+
+  # if we get here, we should have a list of supported fields from TIME
+  # We could reconcile the supported fields against anything the user specified
+  # in the TRILINOS_BUILD_STATS_OUTPUT_FIELDS, which will override the default
+  # fields the parser will output
+  # `time` will actually silently accept bad fields, and give `?field` back
+  # if we use TRILINOS_BUILD_STATS_OUTPUT_FIELDS then bad fields will simply not
+  # be written to a file
+  #
+  # an unimplemented feature in the wrapper is `TRILINOS_BUILD_STATS_PARSE_NM`
+  # which could control if NM is used. (like time, we expect it to work)
+  # we could `find_program` it as well
```

Comment:
R&D guys using their Mac's are the ones I am concerned with. I can realistically see more developers wanting this ON in their daily workflow - `time` is the important one. The NM stuff is `interesting` but it isn't fully baked. For instance, with shared libs you need to look for `exported` variables, yet with static you can just grab `regular` ones.  Performance wise, there's a gain (1.5%) from not using NM.

The NM work is really a proxy for wanting the capability to enable *more* tools. Now that the scripts support the idea of doing more than `time` only - I could use something better than `nm` (such as Google's `Bloaty`) - The ultimate goal is to help foster awareness of code size/symbol counts.

And since we have this capability to chain operations ... we could also do things like parse `include what you use` or even a `tidy` pass from something like Clang-tidy. Reporting in the CSV pass/fail, or leaving more artifacts that could be consumed by a different set of `post processing scripts` (i.e., right now, we have performance related, perhaps in the future we leave code style artifacts that generate some style report)